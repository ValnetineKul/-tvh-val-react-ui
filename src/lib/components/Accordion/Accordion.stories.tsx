import React from 'react';
import { createMeta, createTemplate } from '../../story-utils';
import Typography from '../Typography';
import Accordion from './Accordion';
import Modal from '../../patterns/Dialogs/Modal';
import DialogContent from '../../patterns/Dialogs/DialogContent';
import DialogFooter from '../../patterns/Dialogs/DialogFooter';
import Button from '../Buttons/Button';

type Props = React.ComponentProps<typeof Accordion>;

const items = [
  {
    title: 'About the product',
    content:
      'Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Nulla vitae elit libero, a pharetra augue. Aenean eu leo quam.',
    key: 'about',
  },
  {
    title: 'Technical specs',
    content: (
      <>
        <Typography variant="h4" component="span">
          Title
        </Typography>
        <Typography variant="body500">Another React component is rendered in the AccordionContent</Typography>
      </>
    ),
    key: 'tech_specs',
    badgeCount: 3,
  },
  {
    title: 'Alternatives',
    content: 'Some more regular text',
    key: 'alternatives',
  },
  {
    title: 'Frequently bought together',
    content: 'Some more regular text',
    key: 'bought_together',
  },
  {
    title: 'Frequently bought together and this will be the longest title for example',
    content: 'Some more regular text',
    key: 'long_title',
  },
];

const Template = createTemplate(Accordion);

const ControlledAccordionTemplate = (args: Props) => {
  const [expanded, setExpanded] = React.useState<string[]>([]);

  const { defaultExpandedPanels } = args;
  React.useEffect(() => {
    if (Array.isArray(defaultExpandedPanels)) {
      setExpanded(defaultExpandedPanels);
    }
  }, [defaultExpandedPanels]);

  const handleChange = (panel: string) => (event: React.SyntheticEvent, isExpanded: boolean) => {
    setExpanded(isExpanded ? [panel] : []);
  };

  return <Accordion {...args} expanded={expanded} onChange={handleChange} />;
};

const ControlledWithModalTemplate = (args: Props) => {
  const [expanded, setExpanded] = React.useState<string[]>([]);
  const [newActivePanel, setNewActivePanel] = React.useState<string | undefined>(undefined);
  const [isModalOpen, setIsModalOpen] = React.useState(false);

  const { defaultExpandedPanels } = args;
  React.useEffect(() => {
    if (Array.isArray(defaultExpandedPanels)) {
      setExpanded(defaultExpandedPanels);
    }
  }, [defaultExpandedPanels]);

  const handleModalOpen = () => {
    setIsModalOpen(true);
  };
  const handleModalClose = () => {
    setIsModalOpen(false);
  };

  const handleChange = (panel: string) => (event: React.SyntheticEvent, isExpanded: boolean) => {
    handleModalOpen();
    setNewActivePanel(isExpanded ? panel : undefined);
  };

  const handleSaveClick = () => {
    handleModalClose();
    setExpanded([newActivePanel]);
  };

  const createDialogContent = (key: string | undefined) => {
    if (!key) {
      return 'Do you want to close current accordion panel?';
    }
    const panel = items.find((item) => item.key === key);
    return `Do you want to open "${panel.title}" accordion panel?`;
  };

  return (
    <>
      <Accordion {...args} expanded={expanded} onChange={handleChange} />
      <Modal open={isModalOpen} onClose={handleModalClose} id="controlled-accordion" title="Controlled accordion">
        <DialogContent>{createDialogContent(newActivePanel)}</DialogContent>
        <DialogFooter
          actionButtons={[
            <Button label={`Yes, ${newActivePanel ? 'open' : 'close'}`} variant="primary" onClick={handleSaveClick} />,
            <Button label="Cancel" variant="tertiary" onClick={handleModalClose} />,
          ]}
        />
      </Modal>
    </>
  );
};

export const Primitive = Template.bind({});
Primitive.args = {
  items,
  defaultExpandedPanels: ['alternatives', 'about'],
  multipleOpen: true,
};

Primitive.storyName = 'accordion';
Primitive.argTypes = {
  expanded: {
    table: { disable: true },
  },
  onChange: {
    table: { disable: true },
  },
};

export const ControlledAccordion = ControlledAccordionTemplate.bind({});
ControlledAccordion.args = {
  items,
  multipleOpen: false,
  defaultExpandedPanels: ['bought_together'],
};

ControlledAccordion.storyName = 'controlledAccordion';
ControlledAccordion.argTypes = {
  expanded: {
    control: { disable: true },
  },
  multipleOpen: {
    control: { disable: true },
  },
  onExpandSideEffect: {
    table: { disable: true },
  },
};

export const ControlledWithModal = ControlledWithModalTemplate.bind({});
ControlledWithModal.args = {
  ...ControlledAccordion.args,
};

ControlledWithModal.storyName = 'controlledWithModal';
ControlledWithModal.argTypes = {
  ...ControlledAccordion.argTypes,
};

export default createMeta({
  component: Accordion,
  title: 'Components/Accordions/Accordion',
  argTypes: {
    items: {
      control: {
        disable: true,
      },
    },
    accordionDetailsClassName: {
      table: {
        disable: true,
      },
    },
    buttonProps: {
      table: {
        disable: true,
      },
    },
    titleProps: {
      table: {
        disable: true,
      },
    },
  },
});
