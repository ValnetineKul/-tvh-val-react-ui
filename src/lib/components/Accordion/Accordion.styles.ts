import { makeStyles } from '../../../themes/core';

const useStyles = makeStyles<void, 'expanded'>({ name: 'Accordion' })((theme, _params, classes) => ({
  itemRoot: {
    boxShadow: 'none',
    transition: 'auto',
    '&:before': {
      content: 'none',
    },
  },

  expanded: {
    [`&.${classes.expanded}`]: {
      margin: theme.layoutSpacing['spacing/000'],
    },
  },

  contentRoot: {
    transition: 'auto',
    padding: theme.layoutSpacing['spacing/400'],

    '& > div:first-of-type': {
      width: '100%',
    },
  },

  textContentRoot: {
    maxWidth: `calc(600px + ${theme.layoutSpacing['spacing/500']})`,
  },
}));

export default useStyles;
