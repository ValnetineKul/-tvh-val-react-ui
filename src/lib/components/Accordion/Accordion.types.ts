export type SurfaceColors = '100' | '150' | '200' | '300';
