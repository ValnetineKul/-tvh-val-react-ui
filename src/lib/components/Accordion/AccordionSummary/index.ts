import AccordionSummary from './AccordionSummary';

export type { AccordionSummaryProps } from './AccordionSummary';
export default AccordionSummary;
