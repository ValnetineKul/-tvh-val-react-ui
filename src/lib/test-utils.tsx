import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
