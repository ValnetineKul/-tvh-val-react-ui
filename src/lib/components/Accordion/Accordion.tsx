import type { ReactNode } from 'react';
import React, { useCallback, useEffect } from 'react';
import { default as MuiAccordion } from '@mui/material/Accordion';
import { default as MuiAccordionDetails } from '@mui/material/AccordionDetails';
import Divider from '../Divider';
import Surface from '../Surfaces/Surface';
import AccordionSummary from './AccordionSummary';
import useStyles from './Accordion.styles';
import type { DataAttributes } from '../../types/common';
import type { SurfaceColors } from './Accordion.types';

export interface AccordionProps<PanelKeys> {
  items: {
    title: ReactNode;
    content: ReactNode;
    key: PanelKeys;
    badgeCount?: number;
  }[];
  multipleOpen?: boolean;
  defaultExpandedPanels?: PanelKeys[] | boolean;
  onExpandSideEffect?: (panel: PanelKeys, isExpanded: boolean) => void;
  summarySurfaceColor?: SurfaceColors;
  className?: string;
  buttonProps?: React.HTMLAttributes<HTMLButtonElement> & DataAttributes;
  expanded?: PanelKeys[];
  onChange?: (panel: PanelKeys) => (event: React.SyntheticEvent, isExpanded: boolean) => void;
}

const Accordion = <PanelKeys extends Readonly<string>>({
  items,
  multipleOpen = true,
  className,
  defaultExpandedPanels,
  onExpandSideEffect,
  buttonProps,
  summarySurfaceColor,
  expanded: expandedProp,
  onChange: onChangeProp,
}: AccordionProps<PanelKeys>) => {
  const { classes, cx } = useStyles();

  const isControlled = typeof expandedProp !== 'undefined';
  const [expanded, setExpanded] = React.useState<PanelKeys[]>([]);

  useEffect(() => {
    if (!isControlled) {
      if (!defaultExpandedPanels) return;

      if (defaultExpandedPanels === true) {
        setExpanded(items.map(({ key }) => key));
      } else {
        setExpanded(defaultExpandedPanels);
      }
    }
  }, [defaultExpandedPanels, items, isControlled]);

  const handleChange = useCallback(
    (panel: PanelKeys) => (event: React.SyntheticEvent, isExpanded: boolean) => {
      if (multipleOpen) {
        isExpanded ? setExpanded(expanded.concat(panel)) : setExpanded(expanded.filter((item) => item !== panel));
      } else {
        setExpanded(isExpanded ? [panel] : []);
      }

      if (onExpandSideEffect) {
        onExpandSideEffect(panel, isExpanded);
      }
    },
    [expanded, multipleOpen, onExpandSideEffect]
  );

  const handleChangePanel = (panel: PanelKeys) => {
    if (!isControlled) {
      return handleChange(panel);
    }

    if (onChangeProp) {
      return onChangeProp(panel);
    }
  };

  return (
    <div className={className}>
      {items.map((item) => {
        const panelId = `accordion_panel_${item.key}`;

        const expandedPanels = isControlled ? expandedProp : expanded;
        const isExpanded = expandedPanels.includes(item.key);

        return (
          <MuiAccordion
            expanded={isExpanded}
            onChange={handleChangePanel(item.key)}
            key={item.key}
            classes={{
              root: classes.itemRoot,
              expanded: classes.expanded,
            }}
          >
            <AccordionSummary
              surfaceColor={summarySurfaceColor}
              buttonProps={buttonProps}
              id={panelId}
              expanded={isExpanded}
              title={item.title}
              badgeCount={item.badgeCount}
            />

            <Surface color="150">
              <MuiAccordionDetails
                classes={{
                  root: cx(classes.contentRoot, { [classes.textContentRoot]: typeof item.content === 'string' }),
                }}
              >
                {item.content}
              </MuiAccordionDetails>
              <Divider />
            </Surface>
          </MuiAccordion>
        );
      })}
    </div>
  );
};

export default Accordion;
