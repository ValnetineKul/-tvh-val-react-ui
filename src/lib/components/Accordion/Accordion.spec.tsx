import React from 'react';
import { render, screen, userEvent } from '../../test-utils';
import Accordion from './Accordion';

const accordions = [
  {
    title: 'First accordion',
    content: 'Accordion 1 content',
    key: 'first',
  },
  {
    title: 'Second accordion',
    content: 'Accordion 2 content',
    key: 'second',
  },
];

describe('Accordion', () => {
  it('Should render correctly', () => {
    const { container } = render(<Accordion items={accordions} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should add custom className', () => {
    const { container } = render(<Accordion items={accordions} className="custom-class-name" />);
    expect(container.firstChild).toHaveClass('custom-class-name');
  });

  it('Should keep accordions open', () => {
    render(<Accordion items={accordions} />);
    const firstAccordion = screen.getByRole('button', { name: 'First accordion' });
    const secondAccordion = screen.getByRole('button', { name: 'Second accordion' });

    expect(firstAccordion).toHaveAttribute('aria-expanded', 'false');
    expect(secondAccordion).toHaveAttribute('aria-expanded', 'false');

    firstAccordion.click();
    secondAccordion.click();

    expect(firstAccordion).toBeInTheDocument();
    expect(secondAccordion).toBeInTheDocument();
    expect(firstAccordion).toHaveAttribute('aria-expanded', 'true');
    expect(secondAccordion).toHaveAttribute('aria-expanded', 'true');

    expect(screen.getByLabelText(/First accordion/)).toHaveTextContent('Accordion 1 content');
    expect(screen.getByLabelText(/Second accordion/)).toHaveTextContent('Accordion 2 content');
  });

  it('Should close accordion on 2nd click', () => {
    render(<Accordion items={accordions} />);
    const firstAccordion = screen.getByRole('button', { name: 'First accordion' });

    firstAccordion.click();
    expect(firstAccordion).toHaveClass('Mui-expanded');

    firstAccordion.click();
    expect(firstAccordion).not.toHaveClass('Mui-expanded');
  });

  it('Should close first and open 2nd accordion', () => {
    render(<Accordion items={accordions} multipleOpen={false} />);
    const firstAccordion = screen.getByRole('button', { name: 'First accordion' });
    const secondAccordion = screen.getByRole('button', { name: 'Second accordion' });

    firstAccordion.click();
    secondAccordion.click();

    expect(firstAccordion).not.toHaveClass('Mui-expanded');
    expect(secondAccordion).toHaveClass('Mui-expanded');

    expect(firstAccordion.getAttribute('aria-expanded')).toBe('false');
    expect(secondAccordion.getAttribute('aria-expanded')).toBe('true');
  });

  it('Should expand certain panels by default', () => {
    render(<Accordion items={accordions} defaultExpandedPanels={['first']} />);

    const firstAccordion = screen.getByRole('button', { name: 'First accordion' });
    const secondAccordion = screen.getByRole('button', { name: 'Second accordion' });

    expect(firstAccordion).toHaveAttribute('aria-expanded', 'true');
    expect(secondAccordion).toHaveAttribute('aria-expanded', 'false');
  });

  it('Should expand all panels by default', () => {
    render(<Accordion items={accordions} defaultExpandedPanels />);

    const firstAccordion = screen.getByRole('button', { name: 'First accordion' });
    const secondAccordion = screen.getByRole('button', { name: 'Second accordion' });

    expect(firstAccordion).toHaveAttribute('aria-expanded', 'true');
    expect(secondAccordion).toHaveAttribute('aria-expanded', 'true');
  });

  it('Should call the side effects handler at state changes', () => {
    const handleSideEffects = jest.fn();
    render(<Accordion onExpandSideEffect={handleSideEffects} items={accordions} />);
    const firstAccordion = screen.getByRole('button', { name: 'First accordion' });
    userEvent.click(firstAccordion);

    expect(handleSideEffects).toBeCalledTimes(1);
    expect(handleSideEffects).toBeCalledWith('first', true);
  });

  describe('Controlled component', () => {
    it('Should be controlled', () => {
      render(<Accordion items={accordions} expanded={['first']} />);

      const firstAccordion = screen.getByRole('button', { name: 'First accordion' });
      const secondAccordion = screen.getByRole('button', { name: 'Second accordion' });

      expect(firstAccordion).toHaveAttribute('aria-expanded', 'true');
      expect(secondAccordion).toHaveAttribute('aria-expanded', 'false');
    });

    it('Should call the onChange when controlled', () => {
      const handleChange = jest.fn();
      render(<Accordion expanded={['first']} onChange={handleChange} items={[accordions[0]]} />);
      const firstAccordion = screen.getByRole('button', { name: 'First accordion' });
      userEvent.click(firstAccordion);

      expect(handleChange).toBeCalledTimes(1);
      expect(handleChange).toBeCalledWith('first');
    });
  });
});
