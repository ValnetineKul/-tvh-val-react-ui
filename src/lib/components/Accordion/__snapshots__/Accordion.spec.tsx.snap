// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Should render correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 tss-1t7zslm-Accordion-itemRoot MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters mui-1elwnq4-MuiPaper-root-MuiAccordion-root"
  >
    <button
      aria-controls="accordion_panel_first"
      aria-expanded="false"
      class="MuiButtonBase-root MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters tss-ecggy1-ButtonBase-root-ButtonBase-focusInset-AccordionSummary-summary mui-6a0xca-MuiButtonBase-root-MuiButtonBase-root-MuiAccordionSummary-root"
      id="accordion_panel_first"
      role="button"
      tabindex="0"
      type="button"
    >
      <div
        class="tss-hsuwam-AccordionSummary-content MuiAccordionSummary-content MuiAccordionSummary-contentGutters mui-o4b71y-MuiAccordionSummary-content"
      >
        <div
          class="tss-1wm0ng0-Badge-root-Badge-fullWidth"
        >
          <h4
            class="tss-4-Typography-headerFunctional-ref tss-3eb4ku-Typography-root-Typography-headerFunctional-AccordionSummary-title-Typography-h4"
          >
            First accordion
          </h4>
        </div>
      </div>
      <div
        class="tss-1emsvu8-AccordionSummary-expandIcon MuiAccordionSummary-expandIconWrapper mui-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <div
          class="tss-w1sjks-Icon-root-Icon-sizeMd"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
            color="inherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            plus.svg
          </svg>
        </div>
      </div>
    </button>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth tss-1pam8iy-Divider-root-Divider-horizontal mui-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden mui-bz4dnt-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical mui-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical mui-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="accordion_panel_first"
            class="MuiAccordion-region"
            role="region"
          >
            <div
              class="tss-8l70s6-Surface-root"
            >
              <div
                class="tss-ag2477-Accordion-contentRoot-Accordion-textContentRoot MuiAccordionDetails-root mui-15v22id-MuiAccordionDetails-root"
              >
                Accordion 1 content
              </div>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth tss-1pam8iy-Divider-root-Divider-horizontal mui-9mgopn-MuiDivider-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 tss-1t7zslm-Accordion-itemRoot MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters mui-1elwnq4-MuiPaper-root-MuiAccordion-root"
  >
    <button
      aria-controls="accordion_panel_second"
      aria-expanded="false"
      class="MuiButtonBase-root MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters tss-ecggy1-ButtonBase-root-ButtonBase-focusInset-AccordionSummary-summary mui-6a0xca-MuiButtonBase-root-MuiButtonBase-root-MuiAccordionSummary-root"
      id="accordion_panel_second"
      role="button"
      tabindex="0"
      type="button"
    >
      <div
        class="tss-hsuwam-AccordionSummary-content MuiAccordionSummary-content MuiAccordionSummary-contentGutters mui-o4b71y-MuiAccordionSummary-content"
      >
        <div
          class="tss-1wm0ng0-Badge-root-Badge-fullWidth"
        >
          <h4
            class="tss-4-Typography-headerFunctional-ref tss-3eb4ku-Typography-root-Typography-headerFunctional-AccordionSummary-title-Typography-h4"
          >
            Second accordion
          </h4>
        </div>
      </div>
      <div
        class="tss-1emsvu8-AccordionSummary-expandIcon MuiAccordionSummary-expandIconWrapper mui-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <div
          class="tss-w1sjks-Icon-root-Icon-sizeMd"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
            color="inherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            plus.svg
          </svg>
        </div>
      </div>
    </button>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth tss-1pam8iy-Divider-root-Divider-horizontal mui-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden mui-bz4dnt-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical mui-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical mui-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="accordion_panel_second"
            class="MuiAccordion-region"
            role="region"
          >
            <div
              class="tss-8l70s6-Surface-root"
            >
              <div
                class="tss-ag2477-Accordion-contentRoot-Accordion-textContentRoot MuiAccordionDetails-root mui-15v22id-MuiAccordionDetails-root"
              >
                Accordion 2 content
              </div>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth tss-1pam8iy-Divider-root-Divider-horizontal mui-9mgopn-MuiDivider-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
